src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Build/LanguageServiceErrorListProvider.cs

```diff
@@ -54,6 +69,13 @@ private async Task<AddMessageResult> AddMessageCoreAsync(TargetGeneratedError er
             if (!TryExtractErrorListDetails(error.BuildEventArgs, out ErrorListDetails details) || string.IsNullOrEmpty(details.Code))
                 return AddMessageResult.NotHandled;
 
+            // When this feature flag is enabled, build diagnostics will be published by CPS and should not be passed to roslyn.
```

Comment:
Who de-dups the results in that case?

Reply:
De-duping will be handled by the (newish) diagnostics brokered service on the editor side.  CPS and the LSP client provide diagnostics to the service, it then de-dupes and handles tagging + error list display

Reply:
How does editor know that Roslyn hides certain build errors if they don't appear during background analysis, but shows others? For example, when IntelliSense + Build is selected, it continues to show missing entry point, but hides semantic errors? It used to filter out the semantic errors when we pass it over to them.

Reply:
So essentially what Aditya is doing is de-duping based on diagnostic id.  The intellisense window only shows intellisense tagged errors (produced by LSP).  The build window only shows build errors (tagged as build by CPS).  The build + intellisense shows both, but always prefers diagnostics (based on id) that are produced by the live system (LSP).  

See this doc for more details - https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/23530/Error-List-Diagnostic-API-Platform-level-De-Duplication

Note that we might not get it perfectly correct to start with, so we'll try it out under the experiment and iterate to get it correct.